File: kotlin-sample/src/main/java/examples/ConvertKotlin.java
package examples;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class ConvertKotlin {

    public class User {
        private final String name;
        public User(String name) {
            this.name = name;
        }
    }

    public List<User> createTestList() {
        List<String> names = new ArrayList<>();
        names.add("john");
        names.add("bob");
        names.add("alice");
        return names
                .stream()
                .map(name -> new User(name))
                .collect(Collectors.toList());
    }
}

File: kotlin-sample/src/main/java/examples/ConvertKotlin.kt
package examples

import java.util.*

class ConvertKotlin {

    inner class User(private val name: String)

    fun createTestList(): List<User> {
        val names = ArrayList<String>()
        names.add("john")
        names.add("bob")
        names.add("alice")
        return names
                .map { name -> User(name) }
    }
}
